tests/pipelines/controlnet/test_controlnet.py

```diff
@@ -460,6 +460,26 @@ def test_save_pretrained_raise_not_implemented_exception(self):
             except NotImplementedError:
                 pass
 
+    def test_inference_nested_image_input(self):
+        device = "cpu"
+
+        components = self.get_dummy_components()
+        sd_pipe = StableDiffusionControlNetPipeline(**components)
+        sd_pipe = sd_pipe.to(torch_device)
+        sd_pipe.set_progress_bar_config(disable=None)
+
+        inputs = self.get_dummy_inputs(device)
+        inputs["prompt"] = [inputs["prompt"], inputs["prompt"]]
+        inputs["image"] = [inputs["image"], inputs["image"]]
+        outpus = sd_pipe(**inputs)
+        image = outpus.images
+
+        assert image.shape == (2, 64, 64, 3)
+
+        image_1, image_2 = image
+        # make sure that the outputs are different
+        assert np.sum(np.abs(image_1 - image_2)) > 1e-3
```

Comment:
let's also test the case where we only have one image vs multiple prompts so we have to replicate image across the prompt input

```suggestion
        assert np.sum(np.abs(image_1 - image_2)) > 1e-3
        
        inputs = self.get_dummy_inputs(device)
        inputs["prompt"] = [inputs["prompt"], inputs["prompt"]]
        output_1 = sd_pipe(**inputs)
        
        assert np.abs(image - output_1.images).max() < 1e-3
```